### PR TITLE
[Improve][API] Unified tables_configs and table_list

### PR DESCRIPTION
**Pull request by hawk9821**
### Purpose of this pull request
Currently, the attributes of Seatunnel multi-table configuration are table_list and tables_configs. They are distributed in their respective connectors and are confusing to use. Therefore, it is recommended to unify them.

1. > `table_list `  for structured data sources

2. > `tables_configs`  for unstructured data sources

### Check list

* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  2. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
  3. Add ci label in [label-scope-conf](https://github.com/apache/seatunnel/blob/dev/.github/workflows/labeler/label-scope-conf.yml)
  4. Add e2e testcase in [seatunnel-e2e](https://github.com/apache/seatunnel/tree/dev/seatunnel-e2e/seatunnel-connector-v2-e2e/)
  5. Update connector [plugin_config](https://github.com/apache/seatunnel/blob/dev/config/plugin_config)
* [ ] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

### CODE REVIEW

**Comment by Hisoka-X** [comment]
cc @hailin0

File: seatunnel-api/src/main/java/org/apache/seatunnel/api/table/catalog/CatalogOptions.java [hunk 1]
```diff
@@ -56,4 +58,10 @@ public interface CatalogOptions {
                     .withDescription(
                             "The table names RegEx of the database to capture."
                                     + "The table name needs to include the database name, for example: database_.*\\.table_.*");
+
+    Option<List<Map<String, Object>>> TABLE_LIST =
```
Comment (Hisoka-X) [inline seatunnel-api/src/main/java/org/apache/seatunnel/api/table/catalog/CatalogOptions.java:62]: add some comments for devs about when to use `table_list` or `table_configs`?

**Review by Hisoka-X — COMMENTED** [review]
LGTM except one minior problem. Thanks @hawk9821 !

File: docs/en/concept/schema-feature.md [hunk 2]
```diff
@@ -172,6 +172,46 @@ constraintKeys = [
...
 
+## Multi table schemas
+
+```
+tables_configs = [
+  {
+    schema {
+      table = "database.schema.table1"
+      schema_first = false
+      comment = "comment"
```
Comment (hailin0) [inline docs/en/concept/schema-feature.md:183]: Move `table`, `schema first`, `comment` outside of schema?

Comment (Hisoka-X) [reply docs/en/concept/schema-feature.md:183]: it came from https://github.com/apache/seatunnel/pull/5628, this pr doesn't change it.

**Review by Hisoka-X — APPROVED** [review]
LGTM. cc @hailin0

**Review by hailin0 — APPROVED** [review]

**Pull request merged**